### PR TITLE
[#3049] Upgrade to django 4.2

### LINKED ISSUE #3049: Upgrade to Django 4.2
Checklist to prepare the Django 4.2 LTS upgrade

**3rd party packages to verify/make compatible**

- [x] django-admin-index (should be fine, see https://github.com/maykinmedia/django-admin-index/pull/87)
- [x] django-autoslug (-> update to django-autoslug 1.9.9)
- [x] django-axes (#3218)
- [x] django-better-admin-arrayfield (#3737)
- [x] django-camunda (no changes needed, it's compatible)
- [x] django-capture-tag (outdated but should work)
- [x] django-colorfield
- [x] django-cookie-consent
- [x] django-cors-headers
- [x] django-decorator-include (not maintained, but [appears to work with Django 4.2](https://github.com/twidi/django-decorator-include/pull/92) at least)
- [x] django-digid-eherkenning
- [x] django-hijack
- [x] django-modeltranslation
- [x] django-ordered-model (not in metadata, but appears to work)
- [x] django-privates (v2.0.0 was released to confirm support) 
- [x] django-redis (5.3.0, i.e. thelatest version supports Django 4.2)
- [x] django-solo
- [x] django-tinymce
- [x] django-treebeard
- [x] django-yubin (#3218)
- [x] mozilla-django-oidc-db (confirmed in maykinmedia/mozilla-django-oidc-db#64)
- [x] maykin-django-two-factor-auth[phonenumbers]
- [x] django-timeline-logger
- [x] django-csp
- [x] #3877
- [x] mail-cleaner  https://github.com/maykinmedia/mail-cleaner/pull/5
- [x] djangorestframework~=3.13.0 (upgrade to 3.14)
- [x] drf-nested-routers
- [x] djangorestframework-camel-case
- [x] django-filter
- [x] drf-jsonschema-serializer
- [x] drf-polymorphic
- [x] drf-spectacular
- [x] sentry-sdk[django]
- [x] elastic-apm
- [x] zgw-consumers

**Open Forms itself**

- [x] dark theme for tinymce (https://github.com/open-formulieren/open-forms/pull/3235/commits/d9731f6af09dbefc3b108af93d7efb4d5fef4224)
- [ ] Check deprecation warnings
- [x] Remove `ugettext*` variants (the `u` prefix can be dropped)
- [x] Bump the version of Django's intersphinx version to 4.2
- [x] CSRF_TRUSTED_ORIGINS needs to have the protocol
- [x] Update dark mode handling for TinyMCE? See [here](https://github.com/open-formulieren/open-forms/blob/b6ef8f577bb303f84c79c3910a688b2aa2216fce/src/openforms/js/components/admin/form_design/Editor.js#L12-L14) and [here](https://github.com/open-formulieren/open-forms/blob/b6ef8f577bb303f84c79c3910a688b2aa2216fce/src/openforms/js/initTinymce.js#L35-L37)
- [x] Migrate from `pytz` to `zoneinfo`

### PR DESCRIPTION
**Pull request by SilviaAmAm**
Fixes #3049

Todo:

- [x] Fix datetime test
- [x] Expose dark/light theme control in admin
- [x] Fix light/dark mode integration with tinymce
- [x] Fix styling for tooltip icons
- [x] Update markup React admin components to use the new markup from django templates (brings back the inline-labels)
- [x] Report issues for default project
    - [x] Logout form
    - [x] Missing control for dark/light toggle (admin index + admin login page)
    - [x] Broken tooltip styling
    - [x] Broken admin index style overrides
    - [x] Sort out light/dark styles in proper CSS vars

Old findings:

- Decide what to do about the styling changes:
   - Label no longer on the same line -> checked on a non-modified project, labels are intended to be inline but there may be some responsiveness in play here
   - ~~Submit buttons on the left~~ that's the new django layout, we stick with it :) there are A11Y reasons to do this
Screenshot from 2024-02-09 12-21-43
Screenshot from 2024-02-09 12-22-15

### CODE REVIEW

File: src/openforms/authentication/admin.py [hunk 1]
```diff
@@ -40,6 +40,21 @@ class AuthInfoAdmin(admin.ModelAdmin):
...
+                    ("plugin", "attribute"),
+                    "value",
+                    "submission",
+                    "machtigen",
+                    "loa",
+                    "attribute_hashed",
+                ),
+            },
+        ),
+    )
```
Comment (sergei-maertens) [inline src/openforms/authentication/admin.py:57]: need to revert this

**Comment by SilviaAmAm** [comment]
Seems to be working fine! Just noticed these dark theme things while clicking through. Not sure if they were already a problem before the upgrade:
- The modal is white:
Screenshot from 2024-02-20 10-25-11
- The text of the component categories is not legible:
Screenshot from 2024-02-20 10-25-43
- The JSON preview is in light mode:
Screenshot from 2024-02-20 10-26-35
- The text in hidden fields is not legible:
Screenshot from 2024-02-20 10-30-42

**Comment by sergei-maertens** [comment]
Thanks for testing, these are definitely regressions!

**Comment by sergei-maertens** [comment]
Addressed some dark mode regressions, some of them are present in the current master branch too so I've left those alone to be tackled at some other point.

**Pull request merged**